src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -72,47 +78,91 @@ impl<Src: OpendalSource> SplitReader for OpendalReader<Src> {
+            // For other formats, the corresponding parser will still be used for parsing.
+            if let EncodingProperties::Parquet = &self.parser_config.specific.encoding_config {
+                let record_batch_stream = Box::pin(
+                    ParquetRecordBatchStreamBuilder::new(file_reader)
+                        .await?
+                        .with_batch_size(self.source_ctx.source_ctrl_opts.chunk_size)
+                        .build()?,
+                );
 
-            let data_stream =
-                Self::stream_read_object(self.connector.op.clone(), split, self.source_ctx.clone());
+                #[for_await]
+                for record_batch in record_batch_stream {
+                    let record_batch: RecordBatch = record_batch?;
+                    // Convert each record batch into a stream chunk according to user defined schema.
+                    let chunk: StreamChunk = convert_record_batch_to_stream_chunk(
+                        record_batch,
+                        self.columns.clone(),
+                        split.name.clone(),
+                    )?;
 
-            let parser =
-                ByteStreamSourceParserImpl::create(self.parser_config.clone(), source_ctx).await?;
-            let msg_stream = if need_nd_streaming(&self.parser_config.specific.encoding_config) {
-                parser.into_stream(nd_streaming::split_stream(data_stream))
+                    self.source_ctx
+                        .metrics
+                        .partition_input_count
+                        .with_label_values(&[
+                            &actor_id,
+                            &source_id,
+                            &split_id,
+                            &source_name,
+                            &fragment_id,
+                        ])
+                        .inc_by(chunk.cardinality() as u64);
+                    yield chunk;
+                }
             } else {
-                parser.into_stream(data_stream)
-            };
-            #[for_await]
-            for msg in msg_stream {
-                let msg = msg?;
-                self.source_ctx
-                    .metrics
-                    .partition_input_count
-                    .with_label_values(&[
-                        &actor_id,
-                        &source_id,
-                        &split_id,
-                        &source_name,
-                        &fragment_id,
-                    ])
-                    .inc_by(msg.cardinality() as u64);
-                yield msg;
+                let data_stream =
+                    Self::stream_read_object(file_reader, split, self.source_ctx.clone());
+
+                let parser =
+                    ByteStreamSourceParserImpl::create(self.parser_config.clone(), source_ctx)
+                        .await?;
+                let msg_stream = if need_nd_streaming(&self.parser_config.specific.encoding_config)
+                {
```

Comment:
The main implementation difference between parquet and other source here is for other sources we will create a parser which has `fn into_stream` to convert row based `SourceMessage` stream into a column based `StreamChunk` stream, while for parquet we will convert a column `RecordBatch` based stream into a column based `StreamChunk` stream.

I know it is hard to unify them into the same parser trait. But I think it is cleaner to following the same pattern for parquet source here by:
1. Define a `ParquetSourceParser` struct with `fn into_stream(self, data_stream: ParquetRecordBatchStream) -> impl ChunkSourceStream`
2. Move all the parsing and conversion logic for parquet from `opendal_reader.rs` into the `into_stream` implementation.


In this case, `opendal_reader.rs` will deal with the I/O logic only and the parsing logic will be in the specific parser implementaion, just like before.